src/games/strategy/ui/SwingComponents.java

```diff
@@ -295,6 +297,33 @@ public static JMenu newJMenu(final String menuTitle, final KeyboardCode keyboard
   }
 
   /**
+   * Creates a new component that emulates a multiline label.
+   *
+   * <p>
+   * The multiline label will properly wrap text that has embedded newlines ({@code \n}).
+   * </p>
+   *
+   * @param text The text to be displayed; may be {@code null}.
+   * @param rows The number of rows; must be greater than or equal to zero.
+   * @param cols The number of columns; must be greater than or equal to zero.
```

Comment:
Considering checking the params with guava `Preconditions` (fail early principle)